tools/column_maker/column_maker.xml

```diff
@@ -1,164 +1,170 @@
-            <param name="avoid_scientific_notation" value="yes"/>
-            <output name="out_file1">
-                <assert_contents>
-                    <has_text text="CCDS10397" />
-                    <has_text text=".0000000000001" />
-                    <not_has_text text="1e-13" />
-                </assert_contents>
-            </output>
-        </test>
-    </tests>
-    <help>
-
- .. class:: infomark
-
-**TIP:** If your data is not TAB delimited, use *Text Manipulation-&gt;Convert*
-
------
-
-**What it does**
-
-This tool computes an expression for every row of a dataset and appends the result as a new column (field).
-
-- Columns are referenced with **c** and a **number**. For example, **c1** refers to the first column of a tab-delimited file
-
-- **c3-c2** will add a length column to the dataset if **c2** and **c3** are start and end position
-
------
-
-**Example**
-
-If this is your input::
-
-   chr1  151077881  151077918  2  200  -
-   chr1  151081985  151082078  3  500  +
-
-computing "c4*c5" will produce::
-
-   chr1  151077881  151077918  2  200  -   400.0
-   chr1  151081985  151082078  3  500  +  1500.0
-
-if, at the same time, "Round result?" is set to **YES** results will look like this::
-
-   chr1  151077881  151077918  2  200  -   400
-   chr1  151081985  151082078  3  500  +  1500
-
-You can also use this tool to evaluate expressions. For example, computing "c3>=c2" for Input will result in the following::
-
-   chr1  151077881  151077918  2  200  -  True
-   chr1  151081985  151082078  3  500  +  True
-
-or computing "type(c2)==type('') for Input will return::
-
-   chr1  151077881  151077918  2  200  -  False
-   chr1  151081985  151082078  3  500  +  False
-
-
-The following built-in functions are available::
-
-  abs | all | any | bin | bool | chr | ceil | cmp | complex
-
-  divmod | exp | float | log | log10 | floor | hex | int | len | long
-
-  max | min | oct | ord | pow | range | reversed
-
-  round | sorted | sqrt | str | sum | type | unichr | unicode |
-
-    </help>
-    <citations />
-</tool>
+<tool id="Add_a_column1" name="Compute" version="1.3.1">
+    <description>an expression on every row</description>
+    <requirements>
+        <requirement type="package" version="2.7.13">python</requirement>
+        <requirement type="package" version="4.4">sed</requirement>
+        <requirement type="package" version="1.14">numpy</requirement>
+    </requirements>
+    <command detect_errors="aggressive"><![CDATA[
+        #if $header_lines_conditional.header_lines_select == "yes":
+            (sed -n '1,1p' '$input' | sed  "s|$|%${header_lines_conditional.header_new_column_name}|" | tr "%" "\t") > header &&
+            sed '1,1d' '$input' > data &&
+        #else:
+            touch header &&
+            ln -s '$input' data &&
+        #end if
+
+        ## inject colums and column_types metadata into inputs json
+        #import json
+        #set inputs_dict = json.load(open($inputs))
```

Comment:
interesting, why does the path not need any quotes ...

Reply:
True. Will investigate.

Reply:
Its correct the way it is. Essentially its python passing a string variable 

```
inputs="PATH"
open(inputs)
```

but good comment anyway, because I discovered some missing quotes a few lines below.